### PR TITLE
change url() to re_path() in all required urls.py files

### PR DESCRIPTION
**Pull request by Sparsh-Bansal**
FIXES https://github.com/mozilla/addons-server/issues/15592

In this PR , the old django.conf.urls.url() function is replaced with the new django.urls.re_path() function for routing.
The django.conf.urls.url() function from previous versions is now available as django.urls.re_path(). The old location remains for backwards compatibility, without an imminent deprecation.
Its Reference link is [here](https://docs.djangoproject.com/en/3.1/ref/urls/#url).

* [x] This PR relates to an existing open issue and there are no existing
      PRs open for the same issue.
* [x] Add `Fixes #ISSUENUM` at the top of your PR.
* [x] Add a description of the changes introduced in this PR.
* [x] The change has been successfully run locally.

### CODE REVIEW

File: src/olympia/accounts/urls.py [hunk 1]
```diff
@@ -1,15 +1,17 @@
-from django.conf.urls import include, url
+from django.conf.urls import include
+from django.urls import re_path
 
 from rest_framework.routers import SimpleRouter
 from rest_framework_nested.routers import NestedSimpleRouter
 
-from olympia.bandwagon.views import CollectionAddonViewSet, CollectionViewSet
+from olympia.bandwagon.views import CollectionAddonViewSet, \
+    CollectionViewSet
```
Comment (eviljeff) [inline src/olympia/accounts/urls.py:8]: random extra change here.

Comment (eviljeff) [reply src/olympia/accounts/urls.py:8]: actually, in quite a few places - it seems like you've over-compensated on line length?

Comment (Sparsh-Bansal) [reply src/olympia/accounts/urls.py:8]: I was getting code style test error as in some of the lines , the length was exceeding the 79 characters limit. So I reformatted
the files with limit of 70 characters.

Comment (eviljeff) [reply src/olympia/accounts/urls.py:8]: yes, 79 characters is what we want

Comment (Sparsh-Bansal) [reply src/olympia/accounts/urls.py]: Changes are done with maximum limit of 79 characters.

File: src/olympia/abuse/urls.py [hunk 2]
```diff
@@ -1,15 +1,15 @@
-from django.conf.urls import include, url
+from django.conf.urls import include
+from django.urls import re_path
 
 from rest_framework.routers import SimpleRouter
 from .views import AddonAbuseViewSet, UserAbuseViewSet
 
-
```
Comment (eviljeff) [inline src/olympia/abuse/urls.py:6]: These extra lines aren't enforced by a linting rule but we've decided we like them for consistency (so the imports are always separated by 2 lines from the non-imports). So can you put back all the extra lines at the top in the urls.py files.

**Review by eviljeff — CHANGES_REQUESTED** [review]

File: src/olympia/abuse/urls.py (hunk 2 above)
Comment (eviljeff) [reply src/olympia/abuse/urls.py:6]: I think it's good to go after that.

Comment (Sparsh-Bansal) [reply src/olympia/abuse/urls.py]: Changes are done with line spacing of 2 between imports and non-imports.

**Review by eviljeff — APPROVED** [review]
r+ thanks!

**Comment by caitmuenster** [comment]
Thanks so much for the PR, @Sparsh-Bansal! 🙌 Your contribution has been added to our [recognition wiki](https://wiki.mozilla.org/Add-ons/Contribute/Recognition). 

Welcome onboard! We look forward to seeing you around the project. 😎

**Pull request merged**